webui/src/pages/repositories/repository/fileRenderers/duckdb.tsx

```diff
@@ -94,16 +94,7 @@ export async function runDuckDBQuery(sql: string):  Promise<arrow.Table<any>> {
         // remove registrations
         await Promise.all(fileNames.map(fileName => db.dropFile(fileName)))
     } finally {
-        await closeDuckDBConnection(conn)
-    }
-    return result
-}
-
-async function closeDuckDBConnection(conn: AsyncDuckDBConnection | null) {
```

Comment:
Please check the case where the data is updated outside, while you continue to query the same file from duckdb.
If I remember the code was to handle this case as the cache hold the previous data and the user was getting previous results.

Reply:
Indeed - this is now handled explicitly inside `runDuckDBQuery`, only dropping the files it registered (see line L95)
```typescript
await Promise.all(fileNames.map(fileName => db.dropFile(fileName)))
``` 

tested to work.